app/src/main/java/swati4star/createpdf/adapter/MergeFilesAdapter.java

```diff
@@ -46,12 +48,61 @@ public ViewMergeFilesHolder onCreateViewHolder(@NonNull ViewGroup parent, int vi
 
     @Override
     public void onBindViewHolder(@NonNull ViewMergeFilesHolder holder, int position) {
-        boolean isEncrypted = mPDFUtils.isPDFEncrypted(mFilePaths.get(position));
-        holder.mFileName.setText(FileUtils.getFileName(mFilePaths.get(position)));
-        holder.mEncryptionImage.setVisibility(isEncrypted ? View.VISIBLE : View.INVISIBLE);
+        boolean isPdfFile = isPdfFile(mFilePaths.get(position));
+        if (isPdfFile) {
+            boolean isEncrypted = mPDFUtils.isPDFEncrypted(mFilePaths.get(position));
+            holder.mFileName.setText(FileUtils.getFileName(mFilePaths.get(position)));
+            holder.mEncryptionImage.setVisibility(isEncrypted ? View.VISIBLE : View.INVISIBLE);
+        }
+    }
+
+    // check whether the file is a real pdf file or not
+    public static boolean isPdfFile(String filePath) {
+        String value = getFileHeader(filePath);
+        return value.startsWith("255044462D312E");
```

Comment:
Can you explain this please!

Reply:
maybe the comment should be changed into 'check the file head making sure that the file is encoding in pdf mode'

Reply:
and if it is not in pdf encoding mode, the program will not set the file visible to the user

Reply:
the pdf head “%PDF1.4” in hex is "255044462D312E"